### ISSUE TITLE
Bug: v6 `specifierNameMatches` treats type aliases differently to interfaces

### ISSUE THREAD

**Issue by RebeccaStevens** [typescript-eslint/typescript-eslint#6818]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

utils

### Repro Code

```typescript
function foo(arg1: Bar) {}

type Bar = { bar: string; };
// Or
interface Bar { bar: string; }
```

### Expected Result

`specifierNameMatches` should be able to detect the name of `Bar` regardless of if it was defined with an interface or a type alias.

### Actual Result

It's able to detect the name of the type when it was defined with an interface but not when defined with a type alias.

### Additional Info

To fix, `type.aliasSymbol.escapedName` should be checked in addition to `type.symbol.escapedName`.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `6.0.0-alpha.106` |
| `@typescript-eslint/parser`            | `6.0.0-alpha.106` |
| `@typescript-eslint/scope-manager`     | `6.0.0-alpha.106` |
| `@typescript-eslint/typescript-estree` | `6.0.0-alpha.106` |
| `@typescript-eslint/type-utils`        | `6.0.0-alpha.106` |
| `@typescript-eslint/utils`             | `6.0.0-alpha.106` |
| `TypeScript`                           | `5.0.3` |
| `ESLint`                               | `8.36.0` |
| `node`                                 | `18.15.0` |

**Comment by bradzacher** [comment]
For next time - this would be a better report in terms of the file / exported public API that's broken, rather than the private / internal non-exported function that's broken.
I appreciate the exactness of the report - it's just that without knowledge of the implementation details of the new API it wasn't super easy for me to figure out what you were talking about exactly.

For reference - this is the function in question.
https://github.com/typescript-eslint/typescript-eslint/blob/c9427b78b69f1a6a2453ef2df2be5bf96b7b00bc/packages/type-utils/src/TypeOrValueSpecifier.ts#L119-L128

For a type alias, `type.getSymbol().escapedName === '__type'` whereas for an interface it `=== 'Bar'`.

**Comment by JoshuaKGoldberg** [comment]
Fixed in v6. 👍
